nixos/modules/services/backup/sanoid.nix

```diff
@@ -56,22 +56,12 @@ let
         for allowed values.
       '';
       type = datasetSettingsType;
-    };
-  };
-
-  commonConfig = config: {
-    settings = {
-      hourly = mkDefault config.hourly;
-      daily = mkDefault config.daily;
-      monthly = mkDefault config.monthly;
-      yearly = mkDefault config.yearly;
-      autoprune = mkDefault config.autoprune;
-      autosnap = mkDefault config.autosnap;
+      default = {};
     };
   };
 
   datasetOptions = {
-    useTemplate = mkOption {
+    use_template = mkOption {
```

Comment:
This is a breaking change that needs to be added to the changelog. My personal opinion is that explicitly defined options should use camel case, but I'm not sure if that is the prevailing sentiment.

Maybe we should just get rid of the explicit options altogether and just make users use `settings`?

Reply:
Right, I'm adding a changelog entry. When the option is just a mapping to an upstream option I find it confusing to introduce a different name and introduce a ˋdescriptionˋ to say so. However having an explicit option has a great added value when it comes to type checking. Here, the sophisticated type around ˋuse_templateˋ is quite impressive/useful. So I would keep those explicit options, just maybe rename ˋsettingsˋ to ˋextraConfˋ to clarify its purpose.

Reply:
The `settings` option has specific conventions associated with it, see: https://github.com/NixOS/rfcs/pull/42